backtester/eventhandlers/portfolio/size/size.go

```diff
@@ -67,14 +67,17 @@ func (s *Size) SizeOrder(o order.Event, amountAvailable float64, cs *exchange.Se
 // that is allowed to be spent/sold for an event.
 // As fee calculation occurs during the actual ordering process
 // this can only attempt to factor the potential fee to remain under the max rules
-func (s *Size) calculateBuySize(price, availableFunds, feeRate float64, minMaxSettings config.MinMax) (float64, error) {
+func (s *Size) calculateBuySize(price, availableFunds, feeRate, buyLimit float64, minMaxSettings config.MinMax) (float64, error) {
 	if availableFunds <= 0 {
 		return 0, errNoFunds
 	}
 	if price == 0 {
 		return 0, nil
 	}
 	amount := availableFunds * (1 - feeRate) / price
+	if buyLimit >= minMaxSettings.MinimumSize && buyLimit <= minMaxSettings.MaximumSize && buyLimit <= amount {
+		amount = buyLimit
+	}
```

Comment:
In backtester/eventhandlers/portfolio/portfolio_test.go the last test is failing currently since we don't set a buy limit. So buy limit = 0, minimumSize = 0 and maximumSize = 0 in this test scenario and causes this if condition to be true, setting amount to buylimit which equals 0. And the test is checking to see if the result is not 0. To fix this we could just add another condition in the if statement: buyLimit != 0. Also can you add the same thing to CalculateSellSize function just beneath this one.

Reply:
These changes will also fix up all the tests in size_test.go